ssc/common.cpp

```diff
@@ -1472,15 +1472,24 @@ scalefactors::scalefactors(var_table* v)
 
 std::vector<double> scalefactors::get_factors(const char* name)
 {
-    size_t nyears = vt->as_integer("analysis_period");
+    size_t nyears = 1;
+    if (vt->is_assigned("analysis_period"))
+    {
+        nyears = vt->as_integer("analysis_period");
+    }
     size_t count, i;
-    std::vector<double> scale_factors(nyears); // TODO analysis period
+    std::vector<double> scale_factors(nyears);
     ssc_number_t* parr = vt->as_array(name, &count);
     if (count == 1)
     {
         for (i = 0; i < nyears; i++)
             scale_factors[i] = (ssc_number_t)pow((double)(1 + parr[0] * 0.01), (double)i);
     }
+    else if (count == 0)
```

Comment:
Okay, I see where the length of the array is coming from. So actually I think this 0 count will never happen since there can't be an SSC_ARRAY assigned that has length < 1. In the else branch, you could still have an error if the user provided a `load_escalation` array that is < analysis_period, in which case there should be a message returned to the user.

Reply:
I agree it's rare, but there are some defaults such as `batt_pv_clipping_forecast` that are zero length arrays. Good thinking on the error message!